library/Zend/Http/Client/Adapter/Curl.php

```diff
@@ -228,10 +228,12 @@ public function connect($host, $port = 80, $secure = false)
             curl_setopt($this->_curl, CURLOPT_CONNECTTIMEOUT, $this->_config['timeout']);
         }
 
-        if (defined('CURLOPT_TIMEOUT_MS')) {
-            curl_setopt($this->_curl, CURLOPT_TIMEOUT_MS, $this->_config['timeout'] * 1000);
-        } else {
-            curl_setopt($this->_curl, CURLOPT_TIMEOUT, $this->_config['timeout']);
+        if (array_key_exists('request_timeout', $this->_config)) {
+          if (defined('CURLOPT_TIMEOUT_MS')) {
+              curl_setopt($this->_curl, CURLOPT_TIMEOUT_MS, $this->_config['request_timeout'] * 1000);
+          } else {
+              curl_setopt($this->_curl, CURLOPT_TIMEOUT, $this->_config['request_timeout']);
+          }
```

Comment:
I'm thinking we can make this a bit more readable, and the intention more clear:

```
if (array_key_exists('request_timeout', $this->_config)) {
    $timeout  = $this->config['request_timeout'];
    $constant = CURLOPT_TIMEOUT;
    if (defined('CURLOPT_TIMEOUT_MS')) {
        $timeout *= 1000;
        $constant = constant('CURLOPT_TIMEOUT_MS');
    }
    curl_setopt($this->_curl, $constant, $timeout);
}
```

Also: indentation. :)

I'll merge.